addons/hw_drivers/tools/helpers.py

```diff
@@ -196,15 +197,17 @@ def generate_password():
     """
     Generate an unique code to secure raspberry pi
     """
-    alphabet = 'abcdefghijkmnpqrstuvwxyz23456789'
-    password = ''.join(secrets.choice(alphabet) for i in range(12))
-    shadow_password = crypt.crypt(password, crypt.mksalt())
-    subprocess.call(('sudo', 'usermod', '-p', shadow_password, 'pi'))
+    if main.ssh_service:
```

Comment:
Not sure about this, you can need the userpassword for other things that ssh. If means you are not able to change password without ssh, isn't it weird?

Reply:
As I know, the only purpose of the password of the iot box is to connect with ssh. So it make sense to inform the user that they have to enable ssh in the db before generating a password. If they generate it and try to connect they'll get stuck anyway.
But if there is an other use of the password. Users could indeed need to generate the password

Reply:
What others use of the password do you have in mind ?

Reply:
If you connect a screen and a keyboard on it? Not sure if a real usecase

Reply:
Since it's possible to re-activate ssh from the DB, it will be possible for them to generate a password if they connect a keyboard. Witch is not really a usecase

Reply:
@mart-e Hey, Could you please provide an update or let me know if you need any further information ?